components/push/src/communications.rs

```diff
@@ -209,8 +215,13 @@ impl Connection for ConnectHttp {
             }
         };
 
-        self.uaid = response["uaid"].as_str().map(ToString::to_string);
-        self.auth = response["secret"].as_str().map(ToString::to_string);
+        if self.uaid.is_none() {
+            self.uaid = response["uaid"].as_str().map(ToString::to_string);
+        }
+        if self.auth.is_none() {
+            self.auth = response["secret"].as_str().map(ToString::to_string);
+            println!("### Got Auth {:?}", self.auth);
```

Comment:
Nit: get rid of the println before landing.

Reply:
ah, crap! Thanks!